models/fates/R/model2netcdf.FATES.R

```diff
@@ -11,291 +11,105 @@
 ##' @title Code to convert FATES netcdf output into into CF standard
 ##'
 ##' @param outdir Location of FATES model output
-##' @param sitelat Latitude of the site
-##' @param sitelon Longitude of the site
-##' @param start_date Start time of the simulation
-##' @param end_date End time of the simulation
 ##' 
 ##' @examples  
 ##' \dontrun{
 ##' example.output <- system.file("case.clm2.h0.2004-01-01-00000.nc",package="PEcAn.FATES")
-##' model2netcdf.FATES(outdir="~/",sitelat=42.56341,sitelon=289.1853,start_date="2004-01-01 00:00:00",
-##' end_date="2005-12-31 00:00:00")
+##' model2netcdf.FATES(outdir="~/")
 ##' }
 ##' 
 ##' @export
 ##'
 ##' @author Michael Dietze, Shawn Serbin
-#model2netcdf.FATES <- function(outdir) {
-model2netcdf.FATES <- function(outdir, sitelat, sitelon, start_date, end_date) {
-    
+model2netcdf.FATES <- function(outdir) {
+
     ## Load functions
-    mstmipvar <- PEcAn.utils::mstmipvar
     ncdim_def <- ncdf4::ncdim_def
     ncvar_def <- ncdf4::ncvar_def
     ncatt_get <- ncdf4::ncatt_get
     ncvar_add <- ncdf4::ncvar_add
-    misc.convert <- PEcAn.utils::misc.convert # unit conversions
+#    misc.convert <- PEcAn.utils::misc.convert # unit conversions
     logger.info <- PEcAn.utils::logger.info
     logger.severe <- PEcAn.utils::logger.severe
     logger.warn <- PEcAn.utils::logger.warn
+    
+    #        var_update("AR","AutoResp","kgC m-2 s-1")
+    var_update <- function(ncout,oldname,newname,newunits=NULL){
+      oldunits <- ncdf4::ncatt_get(ncin,oldname,"units")$value
+      if(is.null(newunits)) newunits = oldunits
+      newvar <- ncdf4::ncvar_def(name = newname, units = newunits, dim = xyt)
```

Comment:
Haven't encountered any z dimensions yet, but since this is just a conversion of existing data it would be straightforward to (A) add more dimensions during the definition stage and (B) autodetect the dimensions of the variable being converted.